MetaMask/metamask-extension
# Approvals selector

## Explanation

This is one of the tickets aiming to refactor selectors to use the pending approvals. In this case, it replaces the one triggered with `wallet_watchAsset` method.

This should not affect the app functionality.

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/451

## Manual Testing Steps

- Go to the [Test Dapp](https://metamask.github.io/test-dapp/)
- Scroll to the EIP 747 section
- Try to add an ERC20 Token, it should still work.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
